pop-benchmarker/src/main/java/com/microsoft/testing/popbenchmarker/MainActivity.java

```diff
@@ -0,0 +1,385 @@
+
+    private MainThreadMarshaller<String> mThreadMarshaller = new MainThreadMarshaller<>();
+
+    private List<Long> mKeyGenerationTimings = new ArrayList<>();
+    private List<Long> mKeyLoadTimings = new ArrayList<>();
+    private List<Long> mSigningTimings = new ArrayList<>();
+    //private boolean mAlreadyExecuted = false;
+
+    private TextView
+            mTv_Manufacturer,
+            mTv_Model,
+            mTv_OsVer,
+            mTv_ApiLevel,
+            mTv_KeyGen,
+            mTv_KeyLoad,
+            mTv_Signing,
+            mTv_HardwareIsolated;
+
+    private Button mBtn_Restart;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        initializeViews();
+        setText(mTv_Manufacturer, Build.MANUFACTURER);
+        setText(mTv_Model, Build.MODEL);
+        setText(mTv_OsVer, Build.VERSION.RELEASE);
+        setText(mTv_ApiLevel, Build.VERSION.SDK_INT);
+
+        executeBenchmarks();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+    }
+
+    private void executeBenchmarks() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                final int iterations = 20;
+                int thisIteration = iterations;
+
+                while (thisIteration > 1) {
+                    final int finalThisIteration = thisIteration;
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            mBtn_Restart.setText("Iteration: " + finalThisIteration + " of: " + iterations);
+                        }
+                    });
+                    populateViews();
+                    thisIteration--;
+                }
+
+                computeAverages();
+
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        mBtn_Restart.setText("Restart");
+                        mBtn_Restart.setEnabled(true);
+                    }
+                });
+            }
+        }).start();
+    }
+
+    private void showToast(@NonNull final String msg) {
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                Toast.makeText(MainActivity.this, msg, Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    private void computeAverages() {
+        final long avgKeyGen = computeAvg(mKeyGenerationTimings);
+        final long avgKeyLoad = computeAvg(mKeyLoadTimings);
+        final long avgSign = computeAvg(mSigningTimings);
+        setText(mTv_KeyGen, String.valueOf(avgKeyGen));
+        setText(mTv_KeyLoad, String.valueOf(avgKeyLoad));
+        setText(mTv_Signing, String.valueOf(avgSign));
+    }
+
+    private long computeAvg(List<Long> list) {
```

Comment:
nit: Rename method something more meaningful like `computeTimingAvg`

nit: Rename `list` param to something more useful like `timingList` or `listOfTimes` 

or if this is a generic compute average method (not to do with the timing) then rename variable on line 151 to something else

Reply:
ok I changed a variable name in ab28d0e09b6c6e0550bddd52f9e678ace5b40840